### ISSUE TITLE
Better visual distinction between --pretty --watch recompile results?

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/TypeScript#20379]
**TypeScript Version:**  2.7.0-dev.20171126

While running `tsc --pretty --watch`, there's not much visual distinction between different rounds of failures. Other than a few blank lines on the right half of the screen, they look very similar. When I have a lot of files I'm going through refactor errors through it can be hard to tell what's fixed or fresh.

![image](https://user-images.githubusercontent.com/3335181/33459577-4f4dced2-d5df-11e7-961e-55541b008f52.png)

For comparison, this is what TSLint's `stylish` formatter outputs...

![image](https://user-images.githubusercontent.com/3335181/33459620-825367ba-d5df-11e7-9c10-13bac16747b1.png)

...and this is what gulp outputs:

![image](https://user-images.githubusercontent.com/3335181/33459831-59fc014a-d5e0-11e7-8e9d-43c9723158d8.png)
****

A few suggestions to make this display easier to read (replace "grey" with any non-white color):
* Use grey text for line/column numbers and/or grey+bold text for `TS2451`-style identifiers
* Use grey text for times
* Use grey and/or italic text for `File change detected.` lines
* In the `Compilation complete.` lines:
    * Say whether compilation failed or succeeded
    * Use green or red text appropriately
* Add a CLI option to clear the console when a new run starts

**Comment by mhegazy** [comment]
Would https://github.com/Microsoft/TypeScript/pull/17756 address the issue?

**Comment by JoshuaKGoldberg** [comment]
Console clearing would make things much better!...

...but it would still be nice to have the more readable colors. This issue would just be much less important.

**Comment by mhegazy** [comment]
#17756 is waiting on one small change, if you want to pick it up and get it in would be happy to merge it.

**Comment by mhegazy** [comment]
Also PRs for additional colors for errors message with --pretty are welcomed.

**Comment by JoshuaKGoldberg** [comment]
I'll try to add the colors in ASAP! :)
